src/views/shared/private-route/private-route.view.tsx

```diff
@@ -8,13 +8,21 @@ const PrivateRoute: FC<PropsWithChildren> = ({ children }) => {
   const { connectedProvider } = useProvidersContext();
   const { pathname, search } = useLocation();
 
-  if (!connectedProvider) {
-    return (
-      <Navigate to={routes.login.path} replace state={{ redirectUrl: `${pathname}${search}` }} />
-    );
+  switch (connectedProvider.status) {
+    case "pending":
+    case "loading":
+    case "reloading": {
```

Comment:
What happens if we return `<>{children}</>` in this case? It would definitely look better than a reload when switching the provider.

Reply:
It won't make any difference today since this AsyncTask is never set to `reloading`. Agreed though. What you propose is more correct and might be relevant tomorrow. Done.

Reply:
This is a good point indeed. Because I've seen that in some effects now we are using `isAsyncTaskDataAvailable` to calculate some things, but I think that we should just check the `successful` status but not the `reloading` one to avoid calculating things unnecessary when we use the `reloading` state at some point.

Reply:
Done as agreed in our meeting. Now we use `connectedProvider.status === "successful"` in all the useEffect's.